### PR TITLE
[api-extractor] Fix an issue with @param tag processing in comments.

### PR DESCRIPTION
**Pull request by iclanton**
In TypeScript 3.0, types mentioned in `@param` tags (i.e. `@param {type} - description`) are recognized as `TypeReference` syntax nodes. This change instructs api-extractor to skip over JSDoc comment nodes entirely.

### CODE REVIEW

File: build-tests/api-extractor-test-05/src/DocClass1.ts [hunk 1]
```diff
@@ -22,9 +22,9 @@ export class DocClass1 {
 
   /**
    * This is an overloaded function.
-   * @param x - the number
-   * @param a - the first string
-   * @param b - the second string
+   * @param {x} - the number
```
Comment (pgonzal) [inline build-tests/api-extractor-test-05/src/DocClass1.ts]: >{x}  [](start = 12, length = 4)

Don't use curly braces; in JSDoc they supply parameter type information, and in TSDoc they aren't used at alll

File: apps/api-extractor/src/aedoc/ApiDocumentation.ts [hunk 2]
```diff
@@ -420,8 +420,14 @@ export class ApiDocumentation {
       const descriptionElements: MarkupBasicElement[] = commentTextElements.concat(remainingElements);
       Markup.normalize(descriptionElements);
 
+      // If the paramter name is in brackets, remove them
+      const bracketedNameMatches: string[] | null = name.match(/^\{\s*([^\s]+)\s*\}$/);
+      const normalizedName: string = bracketedNameMatches && bracketedNameMatches.length === 2
+        ? bracketedNameMatches[1]
+        : name;
```
Comment (pgonzal) [inline apps/api-extractor/src/aedoc/ApiDocumentation.ts]: Let's revert this -- it's deleted in the TSDoc branch anyway :-)

File: apps/api-extractor/src/generators/dtsRollup/AstSymbolTable.ts [hunk 3]
```diff
@@ -190,8 +190,11 @@ export class AstSymbolTable {
    * Used by analyze to recursively analyze the entire child tree.
    */
   private _analyzeChildTree(node: ts.Node, governingAstDeclaration: AstDeclaration): void {
-    // is this a reference to another AstSymbol?
     switch (node.kind) {
+      case ts.SyntaxKind.JSDocComment: // Skip JSDoc comments - TS considers @param tags TypeReference nodes
```
Comment (pgonzal) [inline apps/api-extractor/src/generators/dtsRollup/AstSymbolTable.ts]: > // Skip JSDoc comments - TS considers @param tags TypeReference nodes [](start = 38, length = 70)

Can we add a test case for this?

**Comment by iclanton** [comment]
Approved.

**Comment by iclanton** [comment]
pullapprove isn't cooperating, but this is approved - merging in.

**Pull request merged**